confidence/models.py

```diff
@@ -155,13 +163,13 @@ def get(self, path, default=_NoDefault, as_type=None, resolve_references=True):
             # also a KeyError, but this one should bubble to caller
             raise
         except KeyError as e:
-            if default is not _NoDefault:
+            if default is not NoDefault:
                 return default
             else:
                 missing_key = self._separator.join(steps_taken)
                 raise NotConfiguredError(f'no configuration for key {missing_key}', key=missing_key) from e
 
-    def __getattr__(self, attr):
+    def __getattr__(self, attr: str) -> typing.Any:
```

Comment:
Haven't found the right options yet to auto-inherit type annotations for inherited methods (like all of the magic methods below), seems overkill to spec all of the ones that simply do what their super's tell them to, while still requiring any 'new' methods to be typed...

Reply:
Seems that option doesn't exist (yet?). Let's leave it for now.